lib/ui/ui_dart_state.h

```diff
@@ -48,9 +48,9 @@ class UIDartState : public tonic::DartState {
 
   void FlushMicrotasksNow();
 
-  fml::RefPtr<flutter::SkiaUnrefQueue> GetSkiaUnrefQueue() const;
+  fml::WeakPtr<IOManager> GetIOManager() const;
 
-  fml::WeakPtr<GrContext> GetResourceContext() const;
+  fml::RefPtr<flutter::SkiaUnrefQueue> GetSkiaUnrefQueue() const;
```

Comment:
Should we also delete this method and expect callees to go through the IOManager as well? It seems like that should be the direction we go down - but I'm happy to defer this to another issue (and file one to track that) instead of overloading this PR.

Reply:
Ideally we should never dereference `io_manager_` on the UI thread.

Fixing this for `SkiaUnrefQueue` will require a different approach because it is used widely in UI thread APIs that do not run tasks on the IO thread.  One option would be keeping a strong `RefPtr` to the `SkiaUnrefQueue` in `UIDartState`.